motr/m0crate/crate_io.c

```diff
@@ -260,14 +261,23 @@ int cr_io_vector_prep(struct m0_workload_io *cwi,
 	struct m0_bufvec   *attr    = NULL;
 	struct m0_indexvec *index_vec = NULL;
 	struct m0_bitmap    segment_indices;
+	int                 num_unit_per_op;
 
 	index_vec = m0_alloc(sizeof *index_vec);
-	attr = m0_alloc(sizeof *attr);
-	if (index_vec == NULL || attr == NULL)
+	if (index_vec == NULL)
 		goto enomem;
 
-	rc = m0_indexvec_alloc(index_vec, cwi->cwi_bcount_per_op) ?:
-	     m0_bufvec_alloc(attr, cwi->cwi_bcount_per_op, 1);
+	num_unit_per_op = (cwi->cwi_bcount_per_op * cwi->cwi_bs);
```

Comment:
We are doing overallocation of number of units. Please check if we can get unit_size from layout ID and then get accurate number of units, otherwise we can add comment justifying this overallocation